app/controllers/api/v0/comments_controller.rb

```diff
@@ -1,19 +1,19 @@
 module Api
   module V0
     class CommentsController < ApiController
+      respond_to :json
+
       caches_action :index,
                     cache_path: proc { |c| c.params.permit! },
                     expires_in: 10.minutes
-      respond_to :json
 
       caches_action :show,
                     cache_path: proc { |c| c.params.permit! },
                     expires_in: 10.minutes
-      respond_to :json
 
       def index
         article = Article.find(params[:a_id])
-        @comments = Comment.rooted_on(article, "Article").order(score: :desc)
+        @comments = article.comments.roots.includes(:user).select(%i[id processed_html user_id ancestry])
```

Comment:
`.roots` come from [ancestry](https://github.com/stefankroes/ancestry), the result is the same `where(commentable_id: commentable_id, ancestry: nil, commentable_type: commentable_type)` query we were doing manually before.

It basically only selects root comments under an article, their children and grandchildren will be selected in the JSON afterwards, plan to optimize that as well in the future.